opnode/contracts/Makefile

```diff
@@ -1,56 +1,49 @@
 SHELL := /bin/bash
 
-CONTRACTS_PATH := "../../packages/contracts/artifacts/contracts"
+CONTRACTS_PATH := "../../packages/contracts/"
+OPTIMISM_PORTAL := ../../packages/contracts/contracts/L1/OptimismPortal.sol
+L1_BLOCK_INFO := ../../packages/contracts/contracts/L2/L1Block.sol
 
-binding: abi
-	$(eval temp := $(shell mktemp))
 
-	cat abis/DepositFeed.json \
-		| jq -r .bytecode > $(temp)
+bindings: bindings-l1-block-info bindings-optimism-oracle deployed-bin-l1-block-info
 
-	cat abis/DepositFeed.json \
-		| jq .abi \
-		| abigen --pkg deposit \
-		--abi - \
-		--out deposit/deposit_feed_raw.go \
-		--type deposit \
-		--bin $(temp)
-
-	$(eval deployedBytecode := $(shell cat abis/DepositFeed.json | jq -r .deployedBytecode))
-	echo "// Code generated - DO NOT EDIT." > deposit/deposit_feed_deployed.go
-	echo "// This file is a generated binding and any manual changes will be lost." >> deposit/deposit_feed_deployed.go
-	echo "package deposit" >> deposit/deposit_feed_deployed.go
-	echo "var DepositDeployedBin = \"$(deployedBytecode)\""  >> deposit/deposit_feed_deployed.go
-	gofmt -s -w deposit/deposit_feed_deployed.go
-
-	rm $(temp)
-
-	$(eval temp := $(shell mktemp))
-
-	cat abis/L1Block.json \
-		| jq -r .bytecode > $(temp)
 
-	cat abis/L1Block.json \
-		| jq .abi \
-		| abigen --pkg l1block \
-		--abi - \
-		--out l1block/l1_block_info_raw.go \
-		--type l1block \
-		--bin $(temp)
-
-	$(eval deployedBytecode := $(shell cat abis/L1Block.json | jq -r .deployedBytecode))
+# Split up b/c I don't know how to include this step in the L1 Block Info Bindings
+# What is occuring is that the `temp` variable is hard to pull into the `eval`
+deployed-bin-l1-block-info: bindings-l1-block-info
+	$(eval deployedBytecode := $(shell cat bin/l1block_deployed.hex))
 	echo "// Code generated - DO NOT EDIT." > l1block/l1_block_info_deployed.go
 	echo "// This file is a generated binding and any manual changes will be lost." >> l1block/l1_block_info_deployed.go
 	echo "package l1block" >> l1block/l1_block_info_deployed.go
 	echo "var L1blockDeployedBin = \"$(deployedBytecode)\""  >> l1block/l1_block_info_deployed.go
 	gofmt -s -w l1block/l1_block_info_deployed.go
 
-	rm $(temp)
-
-abi:
-	cat $(CONTRACTS_PATH)/L1/DepositFeed.sol/DepositFeed.json \
-		| jq '{abi,bytecode,deployedBytecode}' \
-		> abis/DepositFeed.json
-	cat $(CONTRACTS_PATH)/L2/L1Block.sol/L1Block.json \
-		| jq '{abi,bytecode,deployedBytecode}' \
-		> abis/L1Block.json
+bindings-l1-block-info:
+	$(eval temp := $(shell mktemp -d))
+	solc \
+		--abi $(L1_BLOCK_INFO) \
+		--bin $(L1_BLOCK_INFO) \
+		--combined-json  abi,bin,bin-runtime \
+		--allow-paths $(CONTRACTS_PATH) \
+		-o $(temp)
+	abigen \
+		--combined-json $(temp)/combined.json \
+		--pkg l1block \
+		--type l1block \
+		--out ./l1block/l1_block_info_raw.go
+	cat $(temp)/combined.json | jq -r ".contracts | with_entries(select(.key | match(\"L1Block\")))[] | .\"bin-runtime\"" > bin/l1block_deployed.hex
+
```

Comment:
There is currently nothing enforcing the version of `solc` being used which could result in non-determinism between different machines

Reply:
This is what foundry uses to manage versioning: https://github.com/roynalnaruto/svm-rs

Is there anyway that we can get the combined json from `forge build`? That would abstract away the `solc` version management

Reply:
Don't we have pragmas on the contracts? I had to switch versions of `solc` this morning to get this to work

Reply:
we previously did do something where we would use the `artifacts` from hardhat. My big worry with that is if it doesn't get rebuilt prior to running this so I prefer building from source here.

Reply:
The pragmas must be exact matches for this to be safe, it looks like they generally are except for the output oracle: https://github.com/ethereum-optimism/optimistic-specs/blob/33071761277e642a4cf409c744916b3b4189f843/packages/contracts/contracts/L1/L2OutputOracle.sol#L2

Reply:
There is also a `--solc` flag for `abigen` that lets you pass a path to `solc` and it'll compile + build the bindings. Perhaps we allow for a configurable `solc` path with just using `solc` as a default, this way anybody can build and we aren't opinionated about how the user manages the version of `solc`

Reply:
Ok. I'll switch this just `0.8.10` and create an issue to look into this later.

Reply:
https://github.com/ethereum-optimism/optimistic-specs/issues/368

Reply:
Longer term the solution should be something like `forge bind go` where foundry manages the bindings being up to date